erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -1599,3 +1599,30 @@ def create_invoice_discounting(source_name, target_doc=None):
 	})
 
 	return invoice_discounting
+
+@frappe.whitelist()
+def create_dunning(source_name, target_doc=None):
+	invoice = frappe.get_doc("Sales Invoice", source_name)
+	dunning = frappe.new_doc("Dunning")
```

Comment:
Instead of manually mapping each field, use `mapper` function. That way, all the fields with same name will be mapped automatically. And if a user adds some custom fields on both the document, those will also get mapped.